src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -556,14 +582,15 @@ def _retrieve_sample(root_dir, env, kbase_collection, source_ver, load_ver):
 
         prepared_samples_data.append(doc)
 
+    data_id_sample_id_map = {doc[names.FLD_KBASE_ID]: doc[names.FLD_KB_SAMPLE_ID] for doc in prepared_samples_data}
```

Comment:
Can you describe how `prepared_samples_data` is generated?

Reply:
👍

Reply:
Sorry, I meant just in this PR comment - I know that there's a json file with the sample data, but I'm wondering how it gets processed by the parser - things like are we making an arango key earlier on that we no longer need and is cruft, does the _mtchsel key still get added, stuff like that

Reply:
👍

Reply:
added more comments. It adds 'kbase_ids', 'genome_count' field and removes 'kbase_id' field and regenerates '_key' field using 'sample_id' instead of 'kbase_id'. All other fields from previous step are reserved including '_mtchsle', 'coll', etc.

Reply:
an example output just in case you are interested to see.

`{"kbase_ids": ["69037_10_1", "69037_4_1", "69037_8_1", "69037_2_1", "69037_6_1"], "_key": "c8e7ddaf4d5931d1b16db7b552e3cc8e", "coll": "GROW", "load_ver": "tian_test_taskfarmer", "_mtchsel": [], "sesar:igsn": "IEGRW0006", "sesar:material": "Liquid>aqueous", "sesar:field_name": "surface water", "other_names": "altamaha_2019_sw_WHONDRS-S19S_0010", "sesar:collection_method": "Grab", "sesar:collection_method_description": "surface water filtered through 0.2micron filter; see collection protocol published at https://data.ess-dive.lbl.gov/view/doi:10.15485/1729719", "purpose": "S19S sampling of global rivers for WHONDRS", "latitude": 31.667875, "longitude": -81.840062, "sesar:physiographic_feature_primary": "stream", "sesar:physiographic_feature_name": "Altamaha", "country": "United States", "sesar:field_program_cruise": "WHONDRS", "sesar:collector_chief_scientist": "Christof Meile", "sesar:collection_date": "7/31/19", "sesar:archive_contact_current": "Mikayla Borton", "sample_template": "SESAR", "_geo_spatial": [-81.840062, 31.667875], "kbase_sample_id": "79db2b9e-0581-4a53-9154-5aa6030aed02", "genome_count": 5}
`

Reply:
So is there a prior step that calculates an arango key that can be removed?